### ISSUE TITLE
A structure in a param is causing two "Expected x" problems.

### ISSUE THREAD

**Issue by michaelalandawson** [softwareCobbler/cfc#9]
cflsp sees two issues with the following valid code.  It is centered around the "." in "mailParams.replyTo".  If I remove the period, the problem is no longer displayed.

````
component {
	public static void function sendReceipt(required struct mailParams) {
		param string mailParams.replyTo = "";
	}
}
````

[{
	"resource": "/c:/[redacted]/classes/form/FormService.cfc",
	"owner": "_generated_diagnostic_collection_name_#1",
	"severity": 8,
	"message": "Expected ';'",
	"source": "cfls",
	"startLineNumber": 3,
	"startColumn": 26,
	"endLineNumber": 3,
	"endColumn": 27
}]

[{
	"resource": "/c:/[redacted]/classes/form/FormService.cfc",
	"owner": "_generated_diagnostic_collection_name_#1",
	"severity": 8,
	"message": "Expected '<number>'",
	"source": "cfls",
	"startLineNumber": 3,
	"startColumn": 27,
	"endLineNumber": 3,
	"endColumn": 28
}]

Thanks!

**Comment by softwareCobbler** [comment]
This should be fixed on 1.0.44, released now.

There's a lot of variability between what is valid on adobe vs. what is valid on lucee, these all do-or-don't parse depending on engine

```
// lucee bad, acf ok (lucee implied type/name requires explicit default attr?)
param array foo = [42];
// lucee bad, acf ok (lucee implied type/name required explicit default attr?)
param array foo.bar = [42];
// lucee ok,  acf bad (adobe, implied type/name, cannot have attrs?)
param array foo.bar default = [42];
// lucee ok,  acf ok
param foo = [42];
```

Anyway, we take the easy way out, and try to parse what is "the intended grammar (as we see it...)" rather than the engine-implemented grammar.

tldr; your example should parse on both engines (but doesn't seem to on Lucee in real life? [trycf](https://trycf.com/gist/d7737f3da3e3927925c10eacfaec7671/lucee5?theme=monokai)) so we should not consider it an error on either engine.

**Comment by michaelalandawson** [comment]
Splendid!  Thanks for doing this.
